gctscript/vm/vm.go

```diff
@@ -176,21 +171,18 @@ func (vm *VM) CompileAndRun() {
 			}
 			return
 		}
-		if vm.T < time.Nanosecond {
-			log.Error(log.GCTScriptMgr, "Repeat timer cannot be under 1 nano second")
-			err = vm.Shutdown()
-			if err != nil {
-				log.Errorln(log.GCTScriptMgr, err)
-			}
+		if vm.T >= time.Nanosecond {
+			vm.runner()
 			return
 		}
-		vm.runner()
-	} else {
-		err = vm.Shutdown()
-		if err != nil {
-			log.Error(log.GCTScriptMgr, err)
+
+		if vm.T > 0 {
+			log.Error(log.GCTScriptMgr, "Repeat timer cannot be under 1 nano second")
```

Comment:
You made this a bit weird. You have if`vm.T >= time.Nanosecond {` aka  `vm.T > 0`,  followed by an actual `if vm.T > 0 {` The purpose of the test `TestVMWithRunnerInvalidTimer` was to error on an invalid timer which you've impacted.
Please review your changes

Reply:
Good catch! Yeah, I have half a mind to remove the timer functionality completely in a future PR as we can dictate for loops and sleeps script side anyway.